### PR TITLE
simplify AppVeyor; fixes AppVeyor?

### LINKED ISSUE #76: Use `AppVeyor` to build binaries after each commit
Just like in `{ospsuite}` package.

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
closes #76

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,60 +1,65 @@
-image: Visual Studio 2019
```
Comment (IndrajeetPatil) [inline appveyor.yml:1]: We don't need VS for R packages.

File: appveyor.yml [hunk 2]
```diff
@@ -1,60 +1,65 @@
...
-      $ErrorActionPreference = "Stop"
-      Invoke-WebRequest http://raw.github.com/krlmlr/r-appveyor/master/scripts/appveyor-tool.ps1 -OutFile "..\appveyor-tool.ps1"
-      Import-Module '..\appveyor-tool.ps1'
+  ps: |
+        Get-Date
+        $ErrorActionPreference = "Stop"
 
 install:
-  - ps: Bootstrap
-  - git submodule update --init --recursive
```
Comment (IndrajeetPatil) [inline appveyor.yml:15]: No submodules present in this repo

**Comment by IndrajeetPatil** [comment]
@msevestre AppVeyor build is never triggered for this repository. Do you know why? 

I don't have access to OSP's AppVeyor profile, so not sure what's going on.

**Comment by IndrajeetPatil** [comment]
@msevestre Any thoughts?

File: appveyor.yml (hunk 2 above)
Comment (msevestre) [reply appveyor.yml:15]: I am very confused here. Why do we have appveyor in this repo? I thought everything was handled using a different strategy because it was on CRAN?

**Comment by msevestre** [comment]
I don't understand why we still have appveyor. Obviously what is currently on appveyor.yml is garbage because it was copied out of ospsuite.r probably and does not apply. But since we  are not using appveyor to build...I am not sure why we need this file at all

File: appveyor.yml (hunk 1 above)
Comment (msevestre) [reply appveyor.yml:1]: ok

File: appveyor.yml [hunk 3]
```diff
@@ -1,60 +1,65 @@
-image: Visual Studio 2019
+# Source: https://github.com/krlmlr/r-appveyor/blob/master/appveyor.yml
 
-version: '{build}'
+# DO NOT CHANGE the "init" and "install" sections below
 
 # Download script file from GitHub
 init:
-  - ps: Update-AppveyorBuild -Version "$($env:ospsuite_version).$($env:appveyor_build_version)"
-  - ps: |
```
Comment (msevestre) [inline appveyor.yml:8]: This shuold stay. Why did you remove this?

File: appveyor.yml [hunk 4]
```diff
@@ -1,60 +1,65 @@
-image: Visual Studio 2019
+# Source: https://github.com/krlmlr/r-appveyor/blob/master/appveyor.yml
 
-version: '{build}'
+# DO NOT CHANGE the "init" and "install" sections below
 
 # Download script file from GitHub
 init:
-  - ps: Update-AppveyorBuild -Version "$($env:ospsuite_version).$($env:appveyor_build_version)"
```
Comment (msevestre) [inline appveyor.yml:7]: this can go

File: appveyor.yml (hunk 2 above)
Comment (msevestre) [reply appveyor.yml:15]: Please keep the bootstrap (this is required to build)
The git submodule can be removed, but not because we do not have submodule. It is done for all package on appveyor by default

File: appveyor.yml [hunk 5]
```diff
@@ -1,60 +1,65 @@
...
+  R_VERSION: "4.1.0"
   R_ARCH: x64
-  R_VERSION: "4.0.5"
-  COVERALLS_TOKEN:
-    secure: xIz/WZT0ex3bs/CMBJTzzdXLhl3sqfSqJ3MshlSY03pZKuyYQN7Z1FprVgnlFMUZ
-
+  KEEP_VIGNETTES: true
+  NOT_CRAN: true
 
+version: "$(app_version).{build}"
```
Comment (msevestre) [inline appveyor.yml:23]: ok this is good. Where is app_version defined?

File: appveyor.yml [hunk 6]
```diff
@@ -1,60 +1,65 @@
...
-  - nuget sources add -name centos -source https://ci.appveyor.com/nuget/ospsuite-centos
-  - nuget install packages.config -OutputDirectory packages # cannot use automatic artifact publish because it's not recognized in a C++/CLI project
   - rake "prepare_for_build[%APPVEYOR_BUILD_VERSION%]"
 
-build_script:
-  - travis-tool.sh install_deps
-  - travis-tool.sh r_binary_install curl
-  - Rscript -e "install.packages('covr', repos = 'http://cran.us.r-project.org')"
+cache:
+  - C:\RLibrary
```
Comment (msevestre) [inline appveyor.yml:29]: remove

File: appveyor.yml [hunk 7]
```diff
@@ -73,28 +78,8 @@ artifacts:
   - path: '*.Rcheck\**\*.Rout'
     name: Logs
 
-  - path: "ospsuite_*_ubuntu18.tar.gz"
-    name: Ubuntu18
-
-  - path: "ospsuite_*_centOS7.tar.gz"
-    name: CentOS7
+  - path: '\*_*.tar.gz'
+    name: Bits
```
Comment (msevestre) [inline appveyor.yml:82]: bits? what about you call in linux and the other one win ?

File: appveyor.yml [hunk 8]
```diff
@@ -1,60 +1,65 @@
...
 before_build:
-  - nuget sources add -name utility -source https://ci.appveyor.com/nuget/ospsuite-utility
-  - nuget sources add -name serializer -source https://ci.appveyor.com/nuget/ospsuite-serializer
-  - nuget sources add -name funcparser -source https://ci.appveyor.com/nuget/ospsuite-funcparser
-  - nuget sources add -name simmodel -source https://ci.appveyor.com/nuget/ospsuite-simmodel
-  - nuget sources add -name cvodes -source https://ci.appveyor.com/nuget/ospsuite-simmodel-solver-cvodes
-  - nuget sources add -name core -source https://ci.appveyor.com/nuget/ospsuite-core
-  - nuget sources add -name centos -source https://ci.appveyor.com/nuget/ospsuite-centos
-  - nuget install packages.config -OutputDirectory packages # cannot use automatic artifact publish because it's not recognized in a C++/CLI project
   - rake "prepare_for_build[%APPVEYOR_BUILD_VERSION%]"
```
Comment (msevestre) [inline appveyor.yml:26]: if we are using this, we need submodules

**Review by msevestre — CHANGES_REQUESTED** [review]
THere is a lot that is not correct here

File: appveyor.yml (hunk 5 above)
Comment (IndrajeetPatil) [reply appveyor.yml:24]: Forgot to add it. Fixed it now.

File: appveyor.yml (hunk 6 above)
Comment (IndrajeetPatil) [reply appveyor.yml]: Removed.

File: appveyor.yml (hunk 7 above)
Comment (IndrajeetPatil) [reply appveyor.yml]: Changed to the same as in _ospsuite_.

File: appveyor.yml (hunk 3 above)
Comment (IndrajeetPatil) [reply appveyor.yml:8]: The versioning of _ospsuite.utils_ is quite different from that in rest of the _ospsuite_, right? 

Won't `$ospsuite_version` produce incorrect version?

Comment (msevestre) [reply appveyor.yml:8]: yes that's why I said this can go. (the versionning)
But the rest that was removed (Bootstrap etc..) needs to come back. This is part of the original script (unless the script has changeD)

**Comment by msevestre** [comment]
I also created a project on appveyor for this
https://ci.appveyor.com/project/open-systems-pharmacology-ci/ospsuite-rutils

File: appveyor.yml (hunk 3 above)
Comment (IndrajeetPatil) [reply appveyor.yml:8]: Cool. Added it back in.

Comment (IndrajeetPatil) [reply appveyor.yml:8]: I had changed it to match the original script, which has changed in the meantime. But now I have reverted to the one we have been using.

**Comment by IndrajeetPatil** [comment]
>I also created a project on appveyor for this 

Great! Thanks.

**Comment by msevestre** [comment]
The build was triggered
https://ci.appveyor.com/project/open-systems-pharmacology-ci/ospsuite-rutils

but no binaries for linux it seems. And the version is not updated. (still 999). Is this what you want

File: appveyor.yml [hunk 9]
```diff
@@ -73,28 +72,11 @@ artifacts:
   - path: '*.Rcheck\**\*.Rout'
     name: Logs
 
-  - path: "ospsuite_*_ubuntu18.tar.gz"
+  - path: "ospsuite_*ubuntu18.tar.gz"
     name: Ubuntu18
 
-  - path: "ospsuite_*_centOS7.tar.gz"
+  - path: "ospsuite_*centOS7.tar.gz"
```
Comment (msevestre) [inline appveyor.yml:78]: those are not created fot his package. Does not make sense I believe

**Review by msevestre — CHANGES_REQUESTED** [review]

File: appveyor.yml (hunk 9 above)
Comment (IndrajeetPatil) [reply appveyor.yml]: Should I also remove Ubuntu? Not really needed.

Comment (msevestre) [reply appveyor.yml]: yes. We just need the two version of the package. The one for linux and the one for windows (tar.gz vs zip)
see reporting engine to see how this is created

https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/blob/develop/appveyor.yml
